requirements-dev.txt

```diff
@@ -6,6 +6,7 @@ git+https://github.com/asdf-format/asdf-wcs-schemas
 git+https://github.com/astropy/astropy
 git+https://github.com/spacetelescope/gwcs
 git+https://github.com/yaml/pyyaml.git
+git+https://github.com/python-jsonschema/jsonschema
```

Comment:
We will have to revert this change for now.

Basically, `tox` is installing the packages from `requirements-dev.txt` individually, prior to installing asdf itself. Some of these requirements require ASDF to operate (we are just using them for testing purposes in this case). This means that a published version of asdf is installed first during this install phase, all of which require `jsonschema<4.10`.  However, during this process we are also trying to install the development version of `jsonschema`, which is version `4.16`. Thus pip cannot resolve the versions, which causes the crash. 

Thus we need to publish a version of asdf without the `jsonschema` upper pin, prior to enabling dev testing with it

Reply:
Thanks! I removed jsonschema in [af59c7a](https://github.com/asdf-format/asdf/pull/1203/commits/af59c7a8dd7affaee8852f3fdac4eab2dd8189fe)